pymc_experimental/statespace/models/SARIMAX.py

```diff
@@ -0,0 +1,409 @@
+                "constraints": "Positive",
+            },
+            "sigma_state": {
+                "shape": (self.k_posdef,),
+                "constraints": "Positive",
+            },
+            "ar_params": {
+                "shape": (self.p,),
+                "constraints": "None",
+            },
+            "ma_params": {
+                "shape": (self.q,),
+                "constraints": "None",
+            },
+        }
+
+        for name in self.param_names:
+            info[name]["dims"] = self.param_dims[name]
+
+        return {name: info[name] for name in self.param_names}
+
+    @property
+    def state_names(self):
+        if self.state_structure == "fast":
+            states = ["data"]
+            states += [f"D{i}.data" for i in range(self._k_diffs)]
+            states += [f"state_{i + 1}" for i in range(self.k_states - self._k_diffs - 1)]
+
+        else:
+            states = ["data"]
+            if self.p > 0:
+                states += [f"L{i + 1}.data" for i in range(self.p - 1)]
+            states += ["innovations"]
+            if self.q > 0:
+                states += [f"L{i + 1}.innovations" for i in range(self.q - 1)]
+        return states
+
+    @property
+    def observed_states(self):
+        return [self.state_names[0]]
+
+    @property
+    def shock_names(self):
+        return ["innovation"]
+
+    @property
+    def param_dims(self):
+        coord_map = {
+            "x0": (ALL_STATE_DIM,),
+            "P0": (ALL_STATE_DIM, ALL_STATE_AUX_DIM),
+            "sigma_obs": (OBS_STATE_DIM,),
+            "sigma_state": (OBS_STATE_DIM,),
+            "ar_params": (AR_PARAM_DIM,),
+            "ma_params": (MA_PARAM_DIM,),
+        }
+
+        if not self.measurement_error:
+            del coord_map["sigma_obs"]
+        if self.p == 0:
+            del coord_map["ar_params"]
+        if self.q == 0:
+            del coord_map["ma_params"]
+        if self.stationary_initialization:
+            del coord_map["P0"]
+            del coord_map["x0"]
+
+        return coord_map
+
+    @property
+    def coords(self) -> Dict[str, Sequence]:
+        coords = make_default_coords(self)
+        if self.p > 0:
+            coords.update({AR_PARAM_DIM: list(range(1, self.p + 1))})
+        if self.q > 0:
+            coords.update({MA_PARAM_DIM: list(range(1, self.q + 1))})
+
+        return coords
+
+    def _stationary_initialization(self, mode=None):
+        # Solve for matrix quadratic for P0
+        T = self.ssm["transition"]
+        R = self.ssm["selection"]
+        Q = self.ssm["state_cov"]
+        c = self.ssm["state_intercept"]
+
+        x0 = pt.linalg.solve(pt.identity_like(T) - T, c, assume_a="gen", check_finite=True)
+
+        method = "direct" if (self.k_states < 5) or (mode == "JAX") else "bilinear"
+        P0 = solve_discrete_lyapunov(T, pt.linalg.matrix_dot(R, Q, R.T), method=method)
+
+        return x0, P0
+
+    def make_symbolic_graph(self) -> None:
```

Comment:
Are you using the Hamilton representation or Harvey here? Seeing that you are testing against the implementation in statsmodel I am assuming Harvey? Would be great to add a reference here.

[ref](http://www-stat.wharton.upenn.edu/~stine/stat910/lectures/14_state_space.pdf)

Reply:
Yeah it's Harvey, but you can get Hamilton by using `state_structure = "interpretable"`. I could use the Hamilton/Harvey names (statsmodels does), but I was thinking about something that would help a new user know why to pick one or the other

Reply:
Adding a comment should do, maybe something like "this is equivalent to the parameterization used in `statsmodels.tsa.statespace.sarimax.SARIMAX(..., hamilton_representation=False)` which use the Harvey representation (ref ...)"

Reply:
I updated the docstring to be a bit more clear, and added a reference to Harvey (1989). Turns out I was wrong, I don't have Hamilton here, it's more verbose than that. My idea was to have the maximally performant (Harvey) and maximally interpretable (all states are pure AR lags or MA lags with no mixing). 

I'm not opposed to adding Hamilton and even Akaike, if you think people would be interested in having it.

Reply:
Might be interesting to compare performance, but other than that I see no particular reason to add additional parameterization.

Reply:
I was hoping you would say that : )